# 🚲 Cyclistic Bike-Share Case Study
### Google Data Analytics Capstone Project

**Objective**: Analyze how annual members and casual riders use Cyclistic bikes differently in order to support a new marketing strategy focused on converting casual riders into annual members.

**Data Source**: [Divvy Trip Data](https://divvy-tripdata.s3.amazonaws.com/index.html)  
**Time Period**: January 2025 – July 2025

# 🧑 Stakeholders

* Lily Moreno – Director of Marketing

Responsible for developing marketing campaigns and strategies. She aims to increase annual memberships by converting casual riders into long-term members.

* Cyclistic Marketing Analytics Team

A team of data analysts who collect, clean, and analyze historical bike-share data to provide actionable insights. You are part of this team as a junior data analyst.

* Cyclistic Executive Team

Decision-makers who approve new marketing programs. They require insights backed by clear, professional data and visualizations.

* Cyclistic Customers

**Casual Riders:** Purchase single-ride or full-day passes.

**Annual Members:** Subscribe to yearly membership.

## 🎯 Business task

Understand the behavioral differences between **casual riders** and **annual members** using Cyclistic's historical trip data. The goal is to identify patterns that can inform marketing strategies to convert casual users into long-term members.


## 📊 Data overview
This is a sample of the trip data for January 2025. The key column for segmentation is `member_casual`, indicating whether the rider is a member or casual user.

In [ ]:
%%bigquery
SELECT *
FROM `ecstatic-bounty-469914-n8.cyclistic_data.202501-divvy-tripdata`
LIMIT 10


Query is running:   0%|          |

Downloading:   0%|          |

ride_id rideable_type                       started_at  \
0  76937EC83A4BCED2  classic_bike 2025-01-30 21:13:48.816000+00:00   
1  6DA6C332FD9BE337  classic_bike 2025-01-22 07:58:44.880000+00:00   
2  11AF2FA6BB1B7FD8  classic_bike 2025-01-10 00:39:39.951000+00:00   
3  6AAC978E8D7D8970  classic_bike 2025-01-11 11:46:33.939000+00:00   
4  B474316106CBAA4D  classic_bike 2025-01-04 11:06:17.803000+00:00   
5  5BFF112CCE7FD423  classic_bike 2025-01-19 01:07:33.899000+00:00   
6  1FFD25CBC31CFBA1  classic_bike 2025-01-25 13:05:36.455000+00:00   
7  6A99A61FBF95801B  classic_bike 2025-01-01 13:15:34.068000+00:00   
8  6EF370384EC1315C  classic_bike 2025-01-20 00:46:32.227000+00:00   
9  4AD778C3EA11C561  classic_bike 2025-01-29 13:45:48.181000+00:00   

                          ended_at                      start_station_name  \
0 2025-01-31 22:13:39.169000+00:00                   Halsted St & 104th St   
1 2025-01-23 08:58:41.064000+00:00                  Aberdeen St & 103rd St   
2 2025-01-11 01:39:19.420000+00:00                 Greenwood Ave & 79th St   
3 2025-01-12 12:46:22.498000+00:00                  Exchange Ave & 79th St   
4 2025-01-05 12:06:04.645000+00:00              Stony Island Ave & 71st St   
5 2025-01-20 02:07:28.861000+00:00                   Ashland Ave & 63rd St   
6 2025-01-26 14:05:16.008000+00:00  Griffin Museum of Science and Industry   
7 2025-01-02 14:15:13.745000+00:00             Cottage Grove Ave & 43rd St   
8 2025-01-21 01:46:24.693000+00:00                    Wallace St & 35th St   
9 2025-01-30 14:45:26.215000+00:00                      State St & 33rd St   

  start_station_id end_station_name end_station_id  start_lat  start_lng  \
0            20125             None           None  41.705140 -87.642424   
1            24425             None           None  41.707040 -87.650220   
2              576             None           None  41.751320 -87.595982   
3              580             None           None  41.752115 -87.552007   
4     KA1503000045             None           None  41.766493 -87.586461   
5            16948             None           None  41.779374 -87.664843   
6     KA1503000074             None           None  41.791728 -87.583945   
7     TA1308000023             None           None  41.816499 -87.606582   
8     TA1308000045             None           None  41.830707 -87.641255   
9            13216             None           None  41.834734 -87.625813   

   end_lat  end_lng member_casual  
0      NaN      NaN        casual  
1      NaN      NaN        casual  
2      NaN      NaN        casual  
3      NaN      NaN        casual  
4      NaN      NaN        casual  
5      NaN      NaN        casual  
6      NaN      NaN        member  
7      NaN      NaN        casual  
8      NaN      NaN        casual  
9      NaN      NaN        casual

## 📋 Data processing
### Step 1: Combining monthly tables
**Purpose:** Combine all monthly ride data (January – July 2025) into a single master table.  
This will allow us to work with **one dataset** for cleaning, transformation, and analysis, instead of querying each month separately.

*The new table consists of 2,904,857 rows.*

In [ ]:
%%bigquery
CREATE OR REPLACE TABLE `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_raw` AS
 SELECT * FROM `ecstatic-bounty-469914-n8.cyclistic_data.202501-divvy-tripdata`
 UNION ALL
 SELECT * FROM `ecstatic-bounty-469914-n8.cyclistic_data.202502-divvy-tripdata`
 UNION ALL
 SELECT * FROM `ecstatic-bounty-469914-n8.cyclistic_data.202503-divvy-tripdata`
 UNION ALL
 SELECT * FROM `ecstatic-bounty-469914-n8.cyclistic_data.202504-divvy-tripdata`
 UNION ALL
 SELECT * FROM `ecstatic-bounty-469914-n8.cyclistic_data.202505-divvy-tripdata`
 UNION ALL
 SELECT * FROM `ecstatic-bounty-469914-n8.cyclistic_data.202506-divvy-tripdata`
 UNION ALL
 SELECT * FROM `ecstatic-bounty-469914-n8.cyclistic_data.202507-divvy-tripdata`

Query is running:   0%|          |

Empty DataFrame
Columns: []
Index: []

### Step 2: Data cleaning

### Timestamps check
First we checked that all rides have valid timestamps.

In [ ]:
%%bigquery
SELECT *
FROM `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_raw`
WHERE ended_at < started_at

Query is running:   0%|          |

Downloading: |          |

Empty DataFrame
Columns: [ride_id, rideable_type, started_at, ended_at, start_station_name, start_station_id, end_station_name, end_station_id, start_lat, start_lng, end_lat, end_lng, member_casual]
Index: []

**Result**: 0 rows

Conclusion: All ride timestamps are valid, so we do not need to filter out any invalid rides in the cleaning step.

### Duplicates check
We checked if the assembled table has duplicate records using the following query:

In [ ]:
%%bigquery
SELECT
DISTINCT ride_id
FROM
`ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_raw`

Query is running:   0%|          |

Downloading:   0%|          |

ride_id
0        D8108BE8FD0F2683
1        8A6A45B1F65CB9F8
2        1CF86DE005ED2312
3        048610C63AEEDCDD
4        C58FE026845549E9
...                   ...
2904852  64DD04EC2CFB4DB7
2904853  3A4D0D2E0979B566
2904854  F9D16D561A8FBD9D
2904855  AA339826063E9D4D
2904856  FFB1E1A40268C9CF

[2904857 rows x 1 columns]

**Result**: 2,904,857 rows

Conclusion: the table doesn't contain duplicate rides.

### Trip duration check
Next we checked for outliers (unusually long rides i.e. rides that lasted more than 24 hours).

In [ ]:
%%bigquery
SELECT *
FROM (
  SELECT
   ride_id,
   member_casual,
   TIMESTAMP_DIFF(ended_at, started_at, MINUTE) as trip_duration_mins
  FROM
   `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_raw`
) AS temp_table
WHERE
 trip_duration_mins > 1440
ORDER BY
 trip_duration_mins DESC

Query is running:   0%|          |

Downloading:   0%|          |

ride_id member_casual  trip_duration_mins
0     F7CAE51D367E737B        casual                1574
1     380F66B85DF60806        casual                1559
2     B6C460F053E0E86A        casual                1559
3     D5184CBF80C1DFB5        casual                1559
4     272597A4FFB71C6E        casual                1559
...                ...           ...                 ...
3199  4493DC9C4188A8A7        casual                1447
3200  1AF4887E5E9D5CD3        member                1445
3201  77F767267172EBDD        member                1444
3202  DF4836A96A3EC871        member                1443
3203  C1E6CE23D88E091E        casual                1441

[3204 rows x 3 columns]

**Result**: 3204 rows

We decided to check how many of those rides were made by casual riders.

In [ ]:
%%bigquery
SELECT
 member_casual,
  COUNT(*) AS ride_count
FROM
 `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_raw`
WHERE
 TIMESTAMP_DIFF(ended_at, started_at, MINUTE) > 1440
GROUP BY
 member_casual;

Query is running:   0%|          |

Downloading:   0%|          |

member_casual  ride_count
0        casual        2759
1        member         445

We identified that 2759 (about 86%) of 24-hour+ rides were made by casual riders, suggesting this behavior is more typical among non-members.

This pattern may reflect:
- Day-pass users on long leisure trips
- Tourists holding onto bikes
- Potential misuse or docking issues

To avoid skewing the behavioral comparison between casual riders and members, we chose to **exclude rides longer than 24 hours from further analysis**.

### Creating a clean master table

To prepare the dataset for analysis, we performed several cleaning steps:

1. Removed null values from critical fields such as ride_id, member_casual, started_at, and ended_at.

2. Filtered out extreme outliers: Rides lasting less than 1 minute or longer than 24 hours (1440 minutes) were excluded, as they are likely anomalies or uncommon use cases. These rides represented a very small portion of the data and were heavily skewed toward casual users.

3. Calculated new features:

* trip_duration_mins: Ride duration in minutes

* day_of_week: Numeric day of the week the ride started (1 = Sunday, 7 = Saturday)

* hour_of_day: Hour of the day the ride started (0–23)

These transformations were done using the following SQL query:

In [ ]:
%%bigquery
CREATE OR REPLACE TABLE `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_clean` AS
SELECT
  ride_id,
  member_casual,
  rideable_type,
  start_station_name,
  end_station_name,
  started_at,
  ended_at,
  start_lat,
  start_lng,
  end_lat,
  end_lng,
  TIMESTAMP_DIFF(ended_at, started_at, MINUTE) AS trip_duration_mins,
  EXTRACT(DAYOFWEEK FROM started_at) AS day_of_week,
  EXTRACT(HOUR FROM started_at) AS hour_of_day
FROM `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_raw`
WHERE ride_id IS NOT NULL
  AND member_casual IS NOT NULL
  AND started_at IS NOT NULL
  AND ended_at IS NOT NULL
  AND TIMESTAMP_DIFF(ended_at, started_at, MINUTE) BETWEEN 1 AND 1440;

Query is running:   0%|          |

Empty DataFrame
Columns: []
Index: []

The new clean table consists of 2,826,009 rows. 78848 were removed as a result of the cleaning process.

## 🔎 Data analysis

Before analyzing the key differences between members and casual riders, we began by exploring the overall distribution of these two user types. This provides useful context and helps us understand the proportion of each group within the dataset.

In [ ]:
%%bigquery
SELECT
  member_casual,
  COUNT(*) AS total_rides,
  ROUND(COUNT(*) * 100.0 / SUM(COUNT(*)) OVER (), 1) AS percentage_of_total
FROM
  `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_clean`
GROUP BY
  member_casual

Query is running:   0%|          |

Downloading:   0%|          |

member_casual  total_rides  percentage_of_total
0        casual      1001311                 35.4
1        member      1824698                 64.6

**Members** took 1,824,698 rides, accounting for **64.6%** of all trips.

**Casual riders** took 1,001,311 rides, making up the remaining **35.4%**.

Conclusion:
The majority of Cyclistic bike trips are taken by annual members, indicating a strong existing base of committed users. However, casual riders still represent over one-third of total usage — a significant segment with conversion potential.

### 1. Classic vs. electric bikes
We aggregated the rides by memeber type and included average trip duration using the following query:

In [ ]:
%%bigquery
SELECT
 member_casual,
 rideable_type,
 COUNT(*) AS total_rides,
 ROUND(AVG(trip_duration_mins), 1) as avg_duration
FROM
 `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_clean`
GROUP BY
 member_casual,
 rideable_type
ORDER BY
 member_casual,
 total_rides DESC

**Findings:**  
* Casual riders spend **significantly more time on classic bikes** (29.0 min) compared to electric bikes (13.7 min) — a difference of 15.3 minutes. This suggests they may use them for recreational purposes such as sightseeing or leisure rides.

* Members also spend more time on classic bikes (12.8 min) than electric bikes (10.3 min), though **the difference is smaller** (2.5 minutes). Members show shorter average durations overall, indicating more purposeful trips — possibly commuting or errands.

* For both groups, electric bikes are used more frequently, but ride times are shorter, likely due to their speed and convenience.

* This suggests classic bikes may attract casual riders for longer, experience-focused trips, while members prioritize quick, utility-based transportation.

## 2. Weekdays vs. weekends

We continued our analysis by comparing the rides taken on weekends and weekdays by memeber type:

In [ ]:
%%bigquery
SELECT
 member_casual,
 CASE
  WHEN day_of_week IN (1,7)
  THEN "weekend"
  ELSE "weekday"
  END AS day_type,
 COUNT(*) AS total_rides,
FROM
  `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_clean`
GROUP BY
 day_type,
 member_casual
ORDER BY
 member_casual,
 day_type

**Findings:**

* Members ride much more frequently on weekdays, suggesting commuting or routine usage.

* **Casual riders are more balanced but still lean toward weekday rides**, possibly due to tourists or flexible schedules.

* **While not extreme**, the contrast in weekday/weekend usage supports the idea that **casual riders might use bikes more for leisure**, whereas members use them more for transportation or work commutes.

## 3. Hour of day patterns

Next we composed a query to dtermine the hours of day when members and casual riders are most active:

In [ ]:
%%bigquery
SELECT
  member_casual,
  hour_of_day,
  total_rides
FROM (
  SELECT
    member_casual,
    hour_of_day,
    COUNT(*) AS total_rides,
    RANK() OVER (PARTITION BY member_casual ORDER BY COUNT(*) DESC) AS rank
  FROM `ecstatic-bounty-469914-n8.cyclistic_data.master_table_2025_clean`
  GROUP BY member_casual, hour_of_day
)
WHERE rank <= 5

Query is running:   0%|          |

Downloading:   0%|          |

member_casual  hour_of_day  total_rides
0        casual           17        98517
1        casual           16        88598
2        casual           18        85562
3        casual           15        76045
4        casual           14        68097
5        member           17       198050
6        member           16       171784
7        member           18       153317
8        member            8       131563
9        member           15       119437

Casual riders are most active in the late afternoon (2 pm - 6 pm).

Members also peak in the afternoon and early evening (3 pm - 6 pm), but significant activity is also observed at 8 am.

Conclusion:

Both groups show strong activity in the late afternoon.

**Members** have a morning peak at 8:00, which casual riders do not, highlighting their **commute-focused usage**.

**Casual riders** are primarily active in the later afternoon and evening, **likely reflecting recreational or leisure trips**.

# 📊 Data visualization

To effectively communicate the insights uncovered during the analysis phase, we created a series of visualizations using Tableau. The summary tables were first created in BigQuery based on the cleaned dataset. These tables included key aggregated metrics such as:

* Ride counts by member type

* Average trip durations

* Usage patterns by hour of day and day of week

* Preferences for bike types

We then exported these summary tables to Google Sheets and connected them to Tableau as data sources. In Tableau, we built a dashboard to visually illustrate the key behavioral differences between casual riders and annual members.

Each chart is designed to answer a specific business question and support the stakeholders' goal of increasing annual memberships.

[View the Interactive Tableau Dashboard](https://public.tableau.com/views/Cyclisticcapstoneprojectdashboard/Dashboard1?:language=en-GB&:sid=&:redirect=auth&:display_count=n&:origin=viz_share_link)

# ✅ Recommendations

Based on the behavioral differences observed between casual riders and annual members, we propose the following strategies to encourage casual riders to convert to annual membership:

1. **Target classic bike users with long ride durations**

Casual riders tend to use classic bikes for longer trips, likely for leisure or sightseeing. Cyclistic could:

* Offer discounted annual plans to classic bike users after a certain number of long rides.

* Create marketing campaigns that promote how membership provides better value for frequent long-distance riders.

2. **Introduce weekend membership promotions**

The data shows higher casual rider activity on weekends, suggesting a preference for recreational use. Consider:

* Launching a “Weekend Rider to Member” promotion that offers a trial membership with weekend benefits.

* Sending personalized emails to weekend users highlighting savings with an annual plan.

3. **Leverage peak hour usage for upselling**

Peak usage among casual riders is in the afternoon to early evening (2–6 PM). This can be an opportunity to:

* Display in-app pop-ups or ads during peak hours that highlight the benefits of becoming a member (e.g., unlimited rides, priority access).

* Offer limited-time membership discounts triggered by ride frequency during peak hours.